app/assets/javascripts/controllers/alerts/alerts_list_controller.js

```diff
@@ -0,0 +1,132 @@
+/* global miqHttpInject */
+
+angular.module('alertsCenter').controller('alertsListController',
+  ['$window', 'alertsCenterService', '$interval', '$timeout',
+  function($window, alertsCenterService, $interval, $timeout) {
+    var vm = this;
+
+    vm.alerts = [];
+    vm.alertsList = [];
+
+    function processData(response) {
+      var updatedAlerts = alertsCenterService.convertToAlertsList(response);
+
+      // update display data for the alerts from the current alert settings
+      angular.forEach(updatedAlerts, function(nextUpdate) {
+        matchingAlert = _.find(vm.alerts, function(existingAlert) {
+          return nextUpdate.id === existingAlert.id;
+        });
+
+        if (angular.isDefined(matchingAlert)) {
+          nextUpdate.isExpanded = matchingAlert.isExpanded;
+        }
+      });
+
+      vm.alerts = updatedAlerts;
+      vm.loadingDone = true;
+      vm.filterChange();
+
+      $timeout();
```

Comment:
This was used to get a $digest cycle after the change. Using $scope.$digest will sometimes throw an exception (if already in a digest cycle which sometime it is).